src/System.Text.Json/tests/Serialization/CustomConverterTests.Dictionary.cs

```diff
@@ -86,5 +87,108 @@ public static void CustomDictionaryConverter()
                 Assert.Equal(json, jsonSerialized);
             }
         }
+
+        [Fact]
+        public static void CustomDictionaryConverterContravariant()
+        {
+            const string Json = @"{""Key1"":1,""Key2"":2}";
+
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new IDictionaryConverter(10));
+
+            Dictionary<string, long> dictionary = JsonSerializer.Deserialize<Dictionary<string, long>>(Json, options);
+            Assert.Equal(11, dictionary["Key1"]);
+            Assert.Equal(12, dictionary["Key2"]);
+
+            Assert.Equal(Json, JsonSerializer.Serialize(dictionary, options));
+        }
+
+        [Fact]
+        public static void ClassWithCustomDictionaryConverterTest()
+        {
+            const string Json = @"{""MyInt"":32,""MyDictionary"":{""Key1"":1,""Key2"":2},""MyString"":""Hello""}";
+
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new IDictionaryConverter(10));
+
+            ClassWithCustomDictionaryConverter dictionary = JsonSerializer.Deserialize<ClassWithCustomDictionaryConverter>(Json, options);
+            Assert.Equal(11, dictionary.MyDictionary["Key1"]);
+            Assert.Equal(12, dictionary.MyDictionary["Key2"]);
+            Assert.Equal(32, dictionary.MyInt);
+            Assert.Equal("Hello", dictionary.MyString);
+
+            Assert.Equal(Json, JsonSerializer.Serialize(dictionary, options));
+        }
+
+        private class ClassWithCustomDictionaryConverter
+        {
+            public int MyInt { get; set; }
+            public Dictionary<string, long> MyDictionary { get; set; }
+            public string MyString { get; set; }
+        }
+
+        private class IDictionaryConverter : JsonConverter<IDictionary<string, long>>
```

Comment:
If this is an `IDictionary<string, long>` converter, why is `typeToConvert` concrete `Dictionary<string, long>`?

Reply:
Because in the test we using it for converting concrete dictionary (To have property instance of JsonPropertyInfoNotNullableContravariant class instead of JsonPropertyInfoNotNullable class)